### PR TITLE
Stop CI Tests also creating Devices needed for Simulated Tests

### PR DESCRIPTION
**Pull request by TechPreacher**
- Move Simulated Tests to new project: LoRaWan.SimulatedTest.csproj
- Move all classes shred between CI and Simulated Tests to LoRaWan.Test.Shared.csproj
- Modify IntegrationTestFixture.cs and IntegrationTestFixtureAsserts.cs to be abstract base classes for CI and Sim tests in IntegrationTestFixtureBase.cs and IntegrationTestFixtureBaseAsserts.cs new hosted in  LoRaWan.Test.Shared.csproj
- Create IntegrationTestBaseSim.cs and IntegrationTestBaseCi.cs that inherit from IntegrationTestBase.cs
- Create IntegrationTestFixtureCi.cs and IntegrationTestFixtureSim.cs that inherit from IntegrationTestFixture.cs
- Change IntegrationTestFixtureCi.cs and IntegrationTestFixtureSim.cs to only create/verify the devices required for the respective tests.
- Create different appsettings.json and appsettings.local.json for CI and Sim test projects.
- Adapt all test project to conform to StyleCop rules. Added:
  - LoraKeysManagerFacade.Test
  - LoRaWan.IntegrationTest
  - LoRaWanTest

Fixes: AB#1014

### CODE REVIEW

**Comment by TechPreacher** [comment]
I will add 'fullci' asap.

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaToolsTest/LoRaLibTest.cs [hunk 1]
```diff
@@ -36,7 +36,7 @@ public void TestJoinAccept()
                 2, 3, 25, 128,
```
Comment (fbeltrao) [inline LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaToolsTest/LoRaLibTest.cs]: Please revert the changes to using type instead of var for variable declarations.

File: LoRaEngine/test/LoRaWan.IntegrationTest/LoraArduinoSerial.cs [hunk 2]
```diff
@@ -1,12 +1,15 @@
-﻿/*
+﻿// Copyright (c) Microsoft. All rights reserved.
```
Comment (fbeltrao) [inline LoRaEngine/test/LoRaWan.IntegrationTest/LoraArduinoSerial.cs]: Please revert changes to pascal case in method names. The reason we keep this is that we ported if from a c library, and to make future work easier they should match as much as possible the original.
To avoid StyleCop complaining about it add the required elements in this file.

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaToolsTest/ConversionHelperTest.cs [hunk 3]
```diff
@@ -20,7 +20,7 @@ public class ConversionHelperTest
         [InlineData(new byte[] { 0xF1, 0xF2, 0xF3, 0xF4 }, "F1F2F3F4")]
```
Comment (fbeltrao) [inline LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaToolsTest/ConversionHelperTest.cs]: Please revert the changes to using type instead of var for variable declarations.

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaToolsTest/LoRaLibTest.cs (hunk 1 above)
Comment (TechPreacher) [reply LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaToolsTest/LoRaLibTest.cs]: Done.

File: LoRaEngine/test/LoRaWan.IntegrationTest/LoraArduinoSerial.cs (hunk 2 above)
Comment (TechPreacher) [reply LoRaEngine/test/LoRaWan.IntegrationTest/LoraArduinoSerial.cs]: Done. Disabled the following rules in LoraArduinoSerial.cs:
#pragma warning disable SA1300 // Elements should begin with an uppercase letter
#pragma warning disable SA1313 // Parameters should begin with a lowercase letter

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaToolsTest/ConversionHelperTest.cs (hunk 3 above)
Comment (TechPreacher) [reply LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaToolsTest/ConversionHelperTest.cs]: Done.
